inspirehep/records/marshmallow/literature/common/accelerator_experiment.py

```diff
@@ -16,16 +16,22 @@ class AcceleratorExperimentSchemaV1(Schema):
 
     @pre_dump(pass_many=True)
     def resolve_experiment_records(self, data, many):
-        experiment_records_map = self.get_control_numbers_to_resolved_experiments_map(
-            data
-        )
         if not many:
+            experiment_records_map = self.get_control_numbers_to_resolved_experiments_map(
+                data
+            )
             return self.get_resolved_record_or_experiment(experiment_records_map, data)
+        results = []
+        for record in data:
+            experiment_records_map = self.get_control_numbers_to_resolved_experiments_map(
```

Comment:
it doesn't worth to calculate this for each record, ``get_linked_records_from_field`` yields all the linked records. We can either modify ``get_linked_records_from_field`` to return an ``InspireRecord`` obj or modify ``get_control_numbers_to_resolved_experiments_map`` to handle this case. The cleaner way is to modify ``get_linked_records_from_field`` to return ``InspireRecords``, this will allow us to remove https://github.com/inspirehep/inspirehep/blob/296d4af998f932dc54aea8b3faf98f247328ceab/inspirehep/records/marshmallow/literature/common/reference_item.py#L84-L85 as well